client/landing/stepper/utils/performance-tracking.tsx

```diff
@@ -0,0 +1,53 @@
+import { useLayoutEffect, useMemo } from 'react';
+import { useStore } from 'react-redux';
+import {
+	startPerformanceTracking,
+	stopPerformanceTracking,
+} from 'calypso/lib/performance-tracking/lib';
+
+// Manually start performance tracking.
+// Used in the entry point to start data collection as close as possible to boot time.
+export function startStepperPerformanceTracking( {
+	fullPageLoad = false,
+}: {
+	fullPageLoad?: boolean;
+} ) {
+	startPerformanceTracking( 'stepper', { fullPageLoad } );
+}
+
+// This hook starts performance gathering for the Stepper everytime the flow or step change.
+// The library we're using ignores follow-up `start` calls while there's an inflight tracking
+// report, so this hook is always safe, even though the entry point starts its own report on
+// first render.
+export function useStartStepperPerformanceTracking( flow: string, step: string ) {
+	useMemo( () => {
```

Comment:
This looks like a `useEffect`.

Reply:
It does, but we want it to happen immediately, before render, otherwise we won't be tracking the time it takes to render.

As such, neither `useEffect` nor `useLayoutEffect` are suitable, as far as I know. Is there a better hook for this purpose.

Reply:
You taught me something. Thanks!